frontend/src/lib/hooks/useAnchor.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react'
+
+export function useAnchor(hash: string): void {
+    useEffect(() => {
+        if (hash && document.getElementById(hash.substr(1))) {
+            // Check if there is a hash and if an element with that id exists
+            const element = document.getElementById(hash.substr(1))
+
+            if (!element) {
+                return
+            }
+
+            element.classList.add('highlighted')
+            const elementPosition = element.getBoundingClientRect().top + window.pageYOffset
+            window.scrollTo({
+                top: elementPosition - 50 - 32, // compensate for header & top margin of pages
+                behavior: 'smooth',
+            })
```

Comment:
Would `element.scrollIntoView()` not do the job? Slightly wary of `getBoundingClientRect` + hardcoded margin compensation.

Reply:
Yeah so we had this before and it would not actually get the element into view because of the postiiton:fixed header. Agree the hardcoded number isn't brilliant.